test/regression/solvers/delta_eddington.F90

```diff
@@ -0,0 +1,356 @@
+    logical :: is_val_close
+
+    ! [DEV NOTES] Temporarily check for the fixed output of the C++ solver
+    n_col = size( f90_radiation_fields )
+    n_lev = size( f90_radiation_fields(1)%edr_, 1 )
+    n_wl = size( f90_radiation_fields(1)%edr_, 2 )
+    do i_wl = 1, n_wl
+      do i_lev = 1, n_lev
+        do i_col = 1, n_col
+          is_val_close =                                                     &
+              is_close( cpp_radiation_fields(i_col)%edr_(i_lev,i_wl),        &
+                   real( (i_wl-1) * n_lev * n_col + (i_lev-1) * n_col +      &
+                         (i_col-1) + 42, kind=dk ))
+          ASSERT( is_val_close )
+          is_val_close =                                                     &
+              is_close( cpp_radiation_fields(i_col)%eup_(i_lev,i_wl),        &
+                   real( (i_wl-1) * n_lev * n_col + (i_lev-1) * n_col +      &
+                         (i_col-1) + 93, kind=dk ))
+          ASSERT( is_val_close )
+          is_val_close =                                                     &
+              is_close( cpp_radiation_fields(i_col)%edn_(i_lev,i_wl),        &
+                   real( (i_wl-1) * n_lev * n_col + (i_lev-1) * n_col +      &
+                         (i_col-1) + 52, kind=dk ))
+          ASSERT( is_val_close )
+          is_val_close =                                                     &
+              is_close( cpp_radiation_fields(i_col)%fdr_(i_lev,i_wl),        &
+                   real( (i_wl-1) * n_lev * n_col + (i_lev-1) * n_col +      &
+                         (i_col-1) + 5, kind=dk ))
+          ASSERT( is_val_close )
+          is_val_close =                                                     &
+              is_close( cpp_radiation_fields(i_col)%fup_(i_lev,i_wl),        &
+                   real( (i_wl-1) * n_lev * n_col + (i_lev-1) * n_col +      &
+                         (i_col-1) + 24, kind=dk ))
+          ASSERT( is_val_close )
+          is_val_close =                                                     &
+              is_close( cpp_radiation_fields(i_col)%fdn_(i_lev,i_wl),        &
+                   real( (i_wl-1) * n_lev * n_col + (i_lev-1) * n_col +      &
+                         (i_col-1) + 97, kind=dk ))
+          ASSERT( is_val_close )
+        end do
+      end do
+    end do
+
+  end subroutine compare_radiation_fields
+
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+  ! Copies a 3D C array pointer to a Fortran array
+  subroutine copy_c_array_to_fortran( c_array_ptr, f_array, i_col, n_col,     &
+      n_lev, n_wl )
+
+    type(c_ptr), intent(in) :: c_array_ptr
+    real(dk), intent(inout) :: f_array(:,:)
+    integer, intent(in) :: i_col, n_col, n_lev, n_wl
+
+    real(kind=c_double), pointer :: c_array(:,:,:)
+
+    call c_f_pointer( c_array_ptr, c_array, [n_col, n_lev+1, n_wl] )
+    f_array(:,:) = real( c_array(i_col,:,:), kind=dk )
+
+  end subroutine copy_c_array_to_fortran
```

Comment:
should this go in the utils part of tuv? We seem to need to do this in a few places across musica

Reply:
We could, although I would vote in favor of no new fortran development in TUV-x, just gradually moving things to C++